src/sage/modules/free_module_pseudomorphism.py

```diff
@@ -0,0 +1,224 @@
+        Domain: Ambient free module of rank 2 over the principal ideal domain Integer Ring
+        Codomain: Ambient free module of rank 2 over the principal ideal domain Integer Ring
+        sage: f(V((1, 2)))
+        (-4, 1)
+
+        sage: P.<x> = ZZ[]; deriv = P.derivation()
+        sage: M = P^2
+        sage: f = M.pseudohom([[1, 2*x], [x, 1]], deriv, side="right"); f
+        Free module pseudomorphism defined as left-multiplication by the matrix
+        [  1 2*x]
+        [  x   1]
+        twisted by the derivation d/dx
+        Domain: Ambient free module of rank 2 over the integral domain Univariate Polynomial Ring in x over Integer Ring
+        Codomain: Ambient free module of rank 2 over the integral domain Univariate Polynomial Ring in x over Integer Ring
+        sage: e = M((2*x^2 + 3*x + 1, x^3 + 7*x + 4))
+        sage: f(e)
+        (2*x^4 + 16*x^2 + 15*x + 4, 3*x^3 + 6*x^2 + 8*x + 11)
+        sage: f = M.pseudohom([[1, 2], [1, 1]], deriv)
+        sage: f(e)
+        (x^3 + 2*x^2 + 14*x + 8, x^3 + 7*x^2 + 13*x + 13)
+    """
+    def __init__(self, domain, base_morphism, twist=None, codomain=None, side="left"):
+        """
+        Constructs a pseudomorphism of free modules.
+
+        INPUT:
+            -  ``domain``   - the domain of the pseudomorphism; a free module 
+
+            -  ``base_morphism`` - either a morphism or a matrix defining a
+                                   morphism
+
+            -  ``twist`` - a twisting morphism, this is either a morphism or 
+                           a derivation (default: None)
+
+            -  ``codomain`` - the codomain of the pseudomorphism; a free
+                              module  (default: None)
+
+            - side -- side of the vectors acted on by the matrix 
+                      (default: ``"left"``)
+
+        EXAMPLES::
+
+            sage: F = GF(25); V = F^3; twist = F.frobenius_endomorphism(5)
+            sage: phi = V.pseudohom(matrix(F,3,[1..9]), twist)
+            sage: type(phi)
+            <class 'sage.modules.free_module_pseudomorphism.FreeModulePseudoMorphism'>
+        """
+        from sage.structure.element import is_Matrix
+        if is_Matrix(base_morphism):
+            self.base_morphism = domain.hom(base_morphism, codomain)
+        elif isinstance(base_morphism, Morphism):
+            self.base_morphism = base_morphism
+        else:
+            self.base_morphism = domain.hom(matrix(domain.coordinate_ring(), \
+                                            base_morphism), codomain)
+        self.derivation = None
+        self.twist_morphism = None
+        if isinstance(twist, Morphism):
+            self.twist_morphism = twist
+        elif isinstance(twist, RingDerivation):
+            self.twist_morphism = twist.parent().twisting_morphism()
+            if twist:
+                self.derivation = twist
+            else:
+                self.derivation = None
+        self.side = side
+
+    def __call__(self, x):
```

Comment:
I think you should rather implement `_call_` and let the `__call__` method of the parent class do the coercion stuff.